kcd-berlin-2022-06-30/README.md

```diff
@@ -0,0 +1,62 @@
+# Flatcar Container Linux basic intro demo for KCD Berlin, 2022
+
+YAML config contains
+1. user "caddy" set-up
+2. inline HTML and separate GIF for a static website
+3. Systemd unit for running caddy to serve the file
+4. Deactivate automated reboots
+
+## Provisioning Demo
+
+Download a Flatcar release (NOT the current / most recent one; use an older version to demo updates) from [here](https://www.flatcar.org/releases/).
+Don't forget to also download the accompanying `flatcar_production_qemu.sh`; we'll need it for the demo.
+
+Move the image to a new name to keep a pristine version; copy it to the original name to have a working copy.
+```shell
+mv flatcar_production_qemu_image.img flatcar_production_qemu_image.img.pristine
+cp flatcar_production_qemu_image.img.pristine flatcar_production_qemu_image.img
+```
+
+1. Create ignition from YAML:
+   ```shell
+   cat talk.yaml | docker run --rm -v $(pwd):/files -i ghcr.io/flatcar-linux/ct:latest --files-dir /files  > ignition.json
+   ```
+2. Start flatcar. Kudos to @pothos for the cool port forwarding hack.
+   ```shell
+   ./flatcar_production_qemu.sh -i ignition.json -p 8080-:80,hostfwd=tcp::2222 --nographic
+   ```
+3. Point your browser to [http://localhost:8080](http://localhost:8080).
+4. SSH into the instance
+   ```shell
+   ssh -o "UserKnownHostsFile=/dev/null" -o "StrictHostKeyChecking=no" core@localhost -p 2222
+   ```
+5. Show `/srv/www/html` with the files created, `systemctl status kcd-demo-webserver`, and `docker ps`.
+
+
+## Update demo
+
+Using the instance provisioned above we'll check for updates, stage, and activate (reboot).
+While the default behavior of Flatcar is to reboot automatically, this has been turned off in the Ignition config (see `reboot_strategy` in the YAML file).
+It might be necesssary to re-provision fro scratch (i.e. start from a new copy of the pristine downloaded) depending on the time spent on the instance above since the nstance might have staged the update already.
```

Comment:
```suggestion
It might be necessary to re-provision fro scratch (i.e. start from a new copy of the pristine downloaded) depending on the time spent on the instance above since the instance might have staged the update already.
```